src/pygeosx/pysolver/PySolver.cpp

```diff
@@ -0,0 +1,303 @@
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+// Source includes
+#include "../pygeosx.hpp"
+#include "PySolver.hpp"
+#include "../PyGroup.hpp"
+
+#define VERIFY_NON_NULL_SELF( self ) \
+  PYTHON_ERROR_IF( self == nullptr, PyExc_RuntimeError, "Passed a nullptr as self.", nullptr )
+
+#define VERIFY_INITIALIZED( self ) \
+  PYTHON_ERROR_IF( self->group == nullptr, PyExc_RuntimeError, "The PySolver is not initialized.", nullptr )
+
+
+namespace geosx
+{
+extern std::unique_ptr< GeosxState > g_state;
```

Comment:
I don't really like the `extern`. Could we have a function in `pygeosx.hpp` to return these instead?

Reply:
I tried in a first time but I think I had some issues. I will try this again.